### ISSUE TITLE
Questiones about “cond” in classifier = lambda x, t, cond: cond(classifier_fn(x, t))

### ISSUE THREAD

**Issue by CreamyLong** [shaoshitong/diffusion-model-learning#1]
Hi， Thank you for your great work. The code is ok, but I wonder what is the usage of the "cond" here?

> https://github.com/shaoshitong/diffusion-model-learning/blob/main/demo/classifier_guidance_dpm_solver_demo.py#L80

**Comment by shaoshitong** [comment]
"cond" can be interpreted as a loss function, e.g., cross-entropy loss, which is used in 
![image](https://user-images.githubusercontent.com/85086735/227690488-2fafedcc-d0a4-48f6-9092-6435c9d098a6.png)

A example of "cond" is like:

```pyhon
    def condition_2(x1,y=label):
        loss = F.cross_entropy(x1,y)
        return loss
```
where "y" is input by "label", as a default parameter. Specifically, "y" refers to "label", such as "torch.randint(0,1(x1.shape[0],)).to(x1.device)"

**Comment by shaoshitong** [comment]
> "cond" can be interpreted as a loss function, e.g., cross-entropy loss, which is used in ![image](https://user-images.githubusercontent.com/85086735/227690488-2fafedcc-d0a4-48f6-9092-6435c9d098a6.png)
> 
> A example of "cond" is like:
> 
> ```
>     def condition_2(x1,y=label):
>         loss = F.cross_entropy(x1,y)
>         return loss
> ```
> 
> where "y" is input by "label", as a default parameter. Specifically, "y" refers to "label", such as "torch.randint(0,1(x1.shape[0],)).to(x1.device)"

In this case, "condition" is the function "condition_2", which is entered as a parameter to " model_wrapper"https://github.com/LuChengTHU/dpm-solver/blob/4a82c39c4c32467279103dcdacd175fb1827822c/dpm_solver_pytorch.py#L178.

**Comment by CreamyLong** [comment]
Thanks for your time.
